File: C++/08_기본수학_2/Problem_02.cpp
/* 08단계 [ 기본수학 2 ] */
// 제목 : 소수
// 문제 번호 : 2581

#include <iostream>
using namespace std;

bool prime_number(int n)
{
    if (n < 2)
        return false;
    else
        for (int i = 2; i * i <= n; i++)
            if (n % i == 0)
                return false;
    return true;
}

int main(void)
{
    int M, N;
    int sum = 0, x;
    bool flag = true;
    cin >> M >> N;
    for (int i = M; i <= N; i++)
    {
        if (prime_number(i))
        {
            if (flag)
            {
                x = i;
                flag = false;
            }
            sum += i;
        }
    }
    if (sum)
    {
        cout << sum << endl
             << x << endl;
    }
    else
        cout << -1 << endl;
}

File: Python/08_기본수학_2/Problem_02.py
def prime_number(n):
    if n < 2:
        return False
    else:
        i = 2
        while i*i <= n:
            if n % i == 0:
                return False
            i += 1
    return True


p = []
M = int(input())
N = int(input())

i = M
while i <= N:
    if prime_number(i):
        p.append(i)
    i += 1

if sum(p):
    print(f"{sum(p)}\n{p[0]}")
else:
    print(-1)
